app/components/Calendar.client.tsx

```diff
@@ -1,9 +1,169 @@
-import ToastUICalendar from '@toast-ui/react-calendar';
+import { useCallback, useRef } from "react";
+import TUICalendar from "@toast-ui/react-calendar";
+import { ISchedule, ICalendarInfo, IEventScheduleObject, DateType } from "tui-calendar";
+import "tui-calendar/dist/tui-calendar.css";
+import "tui-date-picker/dist/tui-date-picker.css";
+import "tui-time-picker/dist/tui-time-picker.css";
 
-const Calendar = () => (
-  <section>
-    <ToastUICalendar height="100vh" />
-  </section>
-);
 
-export default Calendar;
+const start = new Date();
+const end = new Date(new Date().setMinutes(start.getMinutes() + 30));
+const schedules: ISchedule[] = [
+  {
+    id: '1',
+    calendarId: '1',
+    title: 'timed event',
+    body: 'TOAST UI Calendar',
+    start: new Date(new Date().setHours(start.getHours() + 1)),
+    end: new Date(new Date().setHours(start.getHours() + 4)),
+    location: 'Meeting Room A',
+    attendees: ['A', 'B', 'C'],
+    category: 'time',
+    state: 'Free',
+    isReadOnly: true,
+    color: '#fff',
+    bgColor: '#ccc'
+  },
+  {
+    calendarId: "1",
+    category: "time",
+    isVisible: true,
+    title: "Study",
+    id: "1",
+    body: "Test",
+    start,
+    end
+  },
+  {
+    calendarId: "1",
+    category: "time",
+    isVisible: true,
+    title: "Meeting",
+    id: "1",
+    body: "Description",
+    start: new Date(new Date().setHours(start.getHours() + 1)),
+    end: new Date(new Date().setHours(start.getHours() + 2))
+  }
+];
+
+const calendars: ICalendarInfo[] = [
+  {
+    id: "1",
+    name: "My Calendar",
+    color: "#ffffff",
+    bgColor: "#9e5fff",
+    dragBgColor: "#9e5fff",
+    borderColor: "#9e5fff"
+  }
+];
+
+const Calendar = () => {
+  const cal = useRef<any>();
+
+  const onClickSchedule = useCallback((e: {schedule: ISchedule}) => {
+    const { calendarId, id } = e.schedule;
+    const el = cal.current.calendarInst.getElement(id, calendarId);
+
+    console.log(e, el.getBoundingClientRect());
+  }, []);
+
+  const onBeforeCreateSchedule = useCallback((scheduleData: any) => {
+    console.log(scheduleData);
+
+    const schedule = {
+      id: String(Math.random()),
+      title: scheduleData.title,
+      isAllDay: scheduleData.isAllDay,
+      start: scheduleData.start,
+      end: scheduleData.end,
+      category: scheduleData.isAllDay ? "allday" : "time",
+      dueDateClass: "",
+      location: scheduleData.location,
+      raw: {
+        class: scheduleData.raw["class"]
+      },
+      state: scheduleData.state
+    };
+
+    cal.current.calendarInst.createSchedules([schedule]);
+  }, []);
+
+  const onBeforeDeleteSchedule = useCallback((res: IEventScheduleObject) => {
+    console.log(res);
+
+    const { id, calendarId } = res.schedule;
+
+    cal.current.calendarInst.deleteSchedule(id, calendarId);
+  }, []);
+
+  const onBeforeUpdateSchedule = useCallback((e : { schedule: ISchedule, changes: any}) => {
```

Comment:
should we add an interface for **changes** ?

Reply:
added

Reply:
Would be updating the type of changes, when we would be working more on onBeforeUpdateSchedule function